Makefile

```diff
@@ -164,14 +163,19 @@ undeploy-agent: ## Undeploy controller from the K8s cluster specified in ~/.kube
 deploy-sample-applicationset:
 	kubectl apply -f ./samples/argocd-applicationset/echo-applicationset.yaml
 
+DEV_TLS_DIR = config/webhook-setup/control/tls
+DEV_TLS_CRT ?= $(DEV_TLS_DIR)/tls.crt
+DEV_TLS_KEY ?= $(DEV_TLS_DIR)/tls.key
+
 .PHONY: dev-tls
-dev-tls:
-	test -s config/webhook-setup/control/tls/tls.crt || openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout config/webhook-setup/control/tls/tls.key -out config/webhook-setup/control/tls/tls.crt -subj "/C=IE/O=Red Hat Ltd/OU=HCG/CN=webhook.172.32.0.2.nip.io" -addext "subjectAltName = DNS:webhook.172.32.0.2.nip.io"
+dev-tls: $(DEV_TLS_CRT) ## Generate dev tls webhook cert if necessary.
+$(DEV_TLS_CRT):
+	openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout $(DEV_TLS_KEY) -out $(DEV_TLS_CRT) -subj "/C=IE/O=Red Hat Ltd/OU=HCG/CN=webhook.172.32.0.1.nip.io" -addext "subjectAltName = DNS:webhook.172.32.0.1.nip.io"
```

Comment:
No real reason to use `test` here, we can just use a non phony target that expects a file to be generated. I see other targets using `test`, but they appear to do other tests such as check the version etc...